034-instanced-pipelines/proposal.md

```diff
@@ -0,0 +1,75 @@
+# Summary
+
+Instanced pipelines group together pipelines which share a common template configured with different ((vars)). They provide a simple two-level hierarchy and automatic archiving of instances which are no longer needed.
+
+# Proposal
+
+Pipelines can be configured with 'instance vars' like so:
+
+```yaml
+fly set-pipeline -p branch --instance-var branch=feature/foo
+```
+
+This will configure a `branch` pipeline, with the `((branch))` var set to `"feature/foo"`.
+
+Instance vars are used as part of the pipeline identifier in the UI and API. There can be multiple instances of a pipeline with the same name:
+
+```yaml
+fly set-pipeline -p branch --instance-var branch=feature/foo
+fly set-pipeline -p branch --instance-var branch=feature/bar
```

Comment:
I think the implicit looping of the configuration here could be supported well in a vars file api too, eg

```yaml
fly set-pipeline -p branch --instance-var-file vars.yml

#! vars.yml
---
branch:
- feature/foo
- feature/bar